application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,262 @@
+        }
+    }
+
+    /**
+     * Creates an embedded message to send with the selection menu
+     *
+     * <p>
+     * </p>
+     *
+     * @param title for the embedded message. nullable {@link String}
+     * @param description for the embedded message. nullable {@link String}
+     * @return the formatted embed {@link MessageEmbed}
+     */
+    private static @NotNull MessageEmbed makeEmbed(@Nullable String title,
+            @Nullable String description) {
+
+        if (title == null) {
+            title = "Select your roles:";
+        }
+
+        return new EmbedBuilder().setTitle(title)
+            .setDescription(description)
+            .setColor(new Color(24, 221, 136))
+            .build();
+    }
+
+    @Override
+    public void onSelectionMenu(@NotNull SelectionMenuEvent event, @NotNull List<String> args) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+
+        Guild guild = Objects.requireNonNull(event.getGuild());
+
+        List<Role> selectedRoles = new ArrayList<>();
+        for (SelectOption selectedOption : Objects.requireNonNull(event.getSelectedOptions())) {
+            Role selectedRole = guild.getRoleById(selectedOption.getValue());
+            if (selectedRole != null && guild.getSelfMember().canInteract(selectedRole)) {
+                selectedRoles.add(selectedRole);
+            }
+        }
+
+        // True if the event option was 'choose'
+        if (event.getMessage().isEphemeral()) {
+
+            SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+            menu.setPlaceholder("Select your roles")
+                .setMaxValues(selectedRoles.size())
+                .setMinValues(0);
+
+            // Add selected options to the menu
+            selectedRoles.forEach(role -> menu.addOption(role.getName(), role.getId()));
+
+            event.getChannel()
+                .sendMessageEmbeds(event.getMessage().getEmbeds().get(0))
+                .setActionRow(menu.build())
+                .queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
+
+            return;
+        }
+
+        List<SelectOption> menuOptions =
+                Objects.requireNonNull(event.getInteraction().getComponent()).getOptions();
+
+
+        for (SelectOption selectedOption : menuOptions) {
+            Role role = guild.getRoleById(selectedOption.getValue());
+
+            if (role == null) {
+                logger.info(
+                        "The {} ({}) role has been removed but is still an option in the selection menu",
+                        selectedOption.getLabel(), selectedOption.getValue());
+                continue;
+            }
+
+            if (selectedRoles.contains(role)) {
+                guild.addRoleToMember(member, role).queue();
+            } else {
+                event.getGuild().removeRoleFromMember(member, role).queue();
+            }
```

Comment:
Instead, use Guild#modifyMemberRoles and add/remove all roles together, if possible